src/cli_parser.cpp

```diff
@@ -195,7 +203,29 @@ void CLIParser::parse(int argc, char** argv) {
         result = std::make_unique<cxxopts::ParseResult>(options->parse(argc, argv));
 
         if (result->unmatched().size()) {
-            this->graphOptionsParser.parse(result->unmatched());
+            // HF pull mode
+            if (result->count("pull")) {
+                if (result->count("task")) {
+                    std::string task = result->operator[]("task").as<std::string>();
+                    if (task == "text_generation") {
+                        this->graphOptionsParser.parse(result->unmatched());
```

Comment:
result = graphOptionsParser.parse()
Then you only need to check for unmatched in this file not in every parser